server/application/application_event_reporter.go

```diff
@@ -116,36 +114,49 @@ func (s *applicationEventReporter) streamApplicationEvents(
 		if strings.Contains(err.Error(), "context deadline exceeded") {
 			return fmt.Errorf("failed to get application tree: %w", err)
 		}
-		// we still need process app even without tree, it is in case if app yaml originally contain error,
-		// we still want show it on codefresh ui and erors that related to it
-		logCtx.WithError(err).Error("failed to get application tree")
+
+		// we still need process app even without tree, it is in case of app yaml originally contain error,
+		// we still want to show it the erorrs that related to it on codefresh ui
+		logCtx.WithError(err).Warn("failed to get application tree, resuming")
+	}
+
+	desiredManifests, err, manifestGenErr := s.getDesiredManifests(ctx, a, logCtx)
+	if err != nil {
+		return err
 	}
 
 	if isChildApp(a) {
+		// will get here only for applications that are managed as a resource by another application
 		parentApp := a.Labels[common.LabelKeyAppInstance]
 
 		parentApplicationEntity, err := s.server.Get(ctx, &application.ApplicationQuery{
 			Name: &parentApp,
 		})
 		if err != nil {
-			return fmt.Errorf("failed to get application event: %w", err)
+			return fmt.Errorf("failed to get parent application entity: %w", err)
 		}
 
 		rs := getAppAsResource(a)
 
-		desiredManifests, _, manifestGenErr := s.getDesiredManifests(ctx, parentApplicationEntity, logCtx)
+		parentDesiredManifests, err, manifestGenErr := s.getDesiredManifests(ctx, parentApplicationEntity, logCtx)
```

Comment:
this is the actual fix - calling `getDesiredManifests` early, before other async calls will give argocd time to actually delete the app resource from the cluster